WcaOnRails/app/webpacker/javascript/edit-events/modals/Qualification.js

```diff
@@ -0,0 +1,179 @@
+import React from 'react'
+
+import events from '../../wca/events.js.erb'
+import formats from '../../wca/formats.js.erb'
+import AttemptResultInput from './AttemptResultInput'
+import DatePicker from "react-datepicker";
+
+import {
+  pluralize,
+  matchResult,
+  attemptResultToString,
+} from './utils'
+import {
+  roundIdToString,
+  parseActivityCode,
+} from '../../wca/wcif-utils'
+
+import "react-datepicker/dist/react-datepicker.css";
+
+function toUTCDate(dateString) {
+  if (!dateString) {
+    return null;
+  }
+  let date = new Date(dateString);
+  return new Date(date.getFullYear(), date.getMonth(), date.getDate());
+}
+
+function eventQualificationToString(wcifEvent, qualification, { short } = {}) {
+  if (!qualification) {
+    return "-";
+  }
+  let dateString = "-";
+  if (qualification.whenDate) {
+    let whenDate = moment(qualification.whenDate).toDate();
+    dateString = whenDate.toISOString().substring(0, 10);
+  }
+  switch (qualification.type) {
+    case "ranking":
+      return `Top ${qualification.level} competitors by ${dateString}`;
+    case "single":
+      return `Single of ${attemptResultToString(qualification.level, wcifEvent.id, short)} by ${dateString}`;
+    case "average":
+      return `Average of ${attemptResultToString(qualification.level, wcifEvent.id, short)} by ${dateString}`;
+  }
+}
```

Comment:
Consider moving these util functions to a separate standalone helper file.
Especially `toUTCDate` seems like a useful function to have available in general (or might even exist somewhere already?)

Reply:
Thanks!